homeassistant/components/shelly/utils.py

```diff
@@ -92,7 +92,12 @@ def get_rest_value_from_path(status, device_class, path: str):
         last_boot = datetime.utcnow() - timedelta(seconds=_attribute_value)
         _attribute_value = last_boot.replace(microsecond=0).isoformat()
 
-    if "new_version" in path:
-        _attribute_value = _attribute_value.split("/")[1].split("@")[0]
+    if "_version" in path:
+        if _attribute_value:
```

Comment:
```suggestion
    if "_version" in path and _attribute_value:
```

Reply:
_attribute value needs to be None only if we are dealing with version not for all other settings so IMHO nested if is the only working approach.

Reply:
So move `_attribute_value = None` to line 94 and remove `else`.

Reply:
but _attribute_value is used for all attributes not only for version :-S

Reply:
We don't need `_attribute_value = None` in line 101 because `_attribute_value` is already `None`. So this is enough:
```python
if "_version" in path and _attribute_value:
    _attribute_value = (
        _attribute_value.split("/")[1].split("@")[0].replace("v", "")
    )
```